crates/node/p2p/src/net/driver.rs

```diff
@@ -141,7 +141,15 @@ impl Network {
                         let selector = |handler: &crate::BlockHandler| {
                             handler.topic(timestamp)
                         };
-                        match self.gossip.publish(selector, Some(block)) {
+                        // TODO: sign the payload
+                        let payload_hash = op_alloy_rpc_types_engine::PayloadHash(block.payload.block_hash());
+                        let payload = OpNetworkPayloadEnvelope {
+                            payload: block.payload,
+                            signature: alloy_primitives::Signature::new(Default::default(), Default::default(), false),
+                            payload_hash,
+                            parent_beacon_block_root: block.parent_beacon_block_root,
+                        };
+                        match self.gossip.publish(selector, Some(payload)) {
```

Comment:
To publish a `OpNetworkPayloadEnvelope`, the `OpExecutionPayloadEnvelope` type received over rpc needs to be signed + a payload hash generated.